### PR TITLE
Diona nymphs eject from dionas if a species change occur

### LINKED ISSUE #18138: Nymphs are stuck in the body of the diona if the diona changes race while they are inside.
**Issue Description**:

Nymphs are permeantly stuck in the body of race changed diona.

**What did you expect to happen**: 

It to be ejected out or able to leave

**What happened instead**:

Nymph hell

**Why is this bad/What are the consequences**:

You are permanently stuck in the body of someone else

**Steps to reproduce the problem**:

1.Enter a diona as a nypmh
2.change the diona's race (i discovered this through mutation toxin.

**When did the problem start happening**:

Probably after nymphs were added

### PR DESCRIPTION
**Pull request by Vi3trice**
## What Does This PR Do
If dionas happen to contain nymphs and change species, those will automatically eject.

This doesn't apply to scooped up nymphs.

## Why It's Good For The Game
Fixes #18138

## Images of changes
https://user-images.githubusercontent.com/80771500/193864860-74cc1935-343a-49d9-a3d8-13eb6ef549a7.mp4

## Testing
Spawned a nymph and diona, and had them merge, then changed the species. Stuffing the nymph into storage or on top of your head won't get affected.

## Changelog
:cl:
fix: Geshalt nymphs automatically get ejected from dionas if they happen to transform into a different species
/:cl:

### CODE REVIEW

File: code/modules/mob/living/carbon/human/human.dm [hunk 1]
```diff
@@ -1278,6 +1278,9 @@
 
 		oldspecies.on_species_loss(src)
 
+	for(var/mob/living/simple_animal/diona/N in contents) // Let nymphs wiggle out
```
Comment (Sirryan2002) [inline code/modules/mob/living/carbon/human/human.dm:1281]: may this should be scoped to the Diona `on_species_loss` proc instead

Comment (Vi3trice) [reply code/modules/mob/living/carbon/human/human.dm:1281]: Any possibility of another species somehow getting dionas otherwise? Just making sure

**Review by Sirryan2002 — APPROVED** [review]
code looks good

File: code/modules/mob/living/carbon/human/species/diona.dm [hunk 2]
```diff
@@ -77,6 +77,9 @@
 	. = ..()
 	H.clear_alert("nolight")
 
+	for(var/mob/living/simple_animal/diona/N in H.contents) // Let nymphs wiggle out
+		N.resist()
```
Comment (farie82) [inline code/modules/mob/living/carbon/human/species/diona.dm:81]:
```suggestion
		N.split()
```
Better to call the actual proc itself instead of `resist`

**Review by farie82 — COMMENTED** [review]
The current implementation of `split` won't spit out unconcious/dead nymphs by the way

**Review by SteelSlayer — APPROVED** [review]

**Pull request merged**